pkg/util/util.go

```diff
@@ -0,0 +1,71 @@
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// This function validates the URL, then tries to fetch the data with retries
+// and then reads and returns the data as []byte
+// Returns an error if the URL is invalid, or fetching the data failed or
+// if reading the response body fails.
+func GetURLData(urlString string, attempts int) ([]byte, error) {
+	// Validate URL
+	_, err := url.ParseRequestURI(urlString)
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL: %v", urlString)
+	}
+
+	// Fetch the URL and store the response body
+	data, err := FetchURLWithRetries(urlString, attempts, 1*time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed fetching data from the URL %v: %s", urlString, err)
+	}
+
+	return data, nil
+}
+
+// Try to fetch the given url string, and make <attempts> attempts at it.
+// Wait for <duration> time between each try.
+// This returns the data from  the response body []byte upon successful fetch
+// The passed URL is not validated, so validate the URL before passing to this
+// function
+func FetchURLWithRetries(url string, attempts int, duration time.Duration) ([]byte, error) {
```

Comment:
@kadel imo, it's a nice to have, and since the code is already there in this PR, we can keep it.
Anyway, it retries only if getting a valid resource fails, so should be fine, WDYT?